pandas/core/generic.py

```diff
@@ -144,7 +144,9 @@ def _single_replace(self, to_replace, method, inplace, limit):
     if values.dtype == orig_dtype and inplace:
         return
 
-    result = pd.Series(values, index=self.index, dtype=self.dtype).__finalize__(self)
+    result = pd.Series(values, index=self.index, dtype=self.dtype).__finalize__(
+        self, method="_single_replace"
+    )
```

Comment:
can this be broken into two lines?  probably also should use _constructor instead of pd.Series